src/cmd/migrate.ts

```diff
@@ -0,0 +1,116 @@
+import { diff } from 'deep-diff'
+import { cloneDeep, get, set, unset } from 'lodash'
+import { compare, valid } from 'semver'
+import { Argv } from 'yargs'
+import { prepareEnvironment } from '../common/cli'
+import { OtomiDebugger, terminal } from '../common/debug'
+import { hfValues } from '../common/hf'
+import { getFilename, loadYaml, rootDir } from '../common/utils'
+import { writeValues } from '../common/values'
+import { BasicArguments, setParsedArgs } from '../common/yargs'
+import { askYesNo } from '../common/zx-enhance'
+import { validateValues } from './validate-values'
+
+const cmdName = getFilename(__filename)
+const debug: OtomiDebugger = terminal(cmdName)
+
+interface Change {
+  version: string
+  deletions?: Array<string>
+  locations?: Array<{
+    [oldLocation: string]: string
+  }>
+  mutations?: Array<{
+    [preMutation: string]: string[]
```

Comment:
this is not accomodating the requirements, which states that a mutation is to be expressed with a go template, and as such the interface should be just: 
```suggestion
    [preMutation: string]: string
```
with the right hand side being the template. So the old value on the path (left hand side) should be passed into the template to get the result